pkg/reconcilermanager/controllers/reconciler_base.go

```diff
@@ -105,6 +105,9 @@ type reconcilerBase struct {
 
 	// syncKind is the kind of the sync object: RootSync or RepoSync.
 	syncKind string
+
+	// controllerName is used by tests to de-dupe controllers
+	controllerName string
```

Comment:
Looks like this needs to be populated by the constructors so it's non-empty.

Reply:
The default value for a string is empty string (`""`)

Reply:
Right, but then the name will be empty and cause the requirement to fail, right? I'm not sure how it fails or if it's a test-only requirement, but if we have it here in the struct we should populate it.

Reply:
The controller-runtime requirement is that the name is unique, not that it's non-empty. If you don't explicitly set the name then it defaults to the name of the resource (`rootsync`/`reposync`). The problem with the unit tests is that they always use the default name.

Reply:
Gotcha. Then we probably don't need to require the caller to set them, but it might make sense to hard-code the name in the constructor so it's always set. Otherwise it looks like we have a test-only field in a non-test struct.

Reply:
Alternatively, we could set it to the syncKind field value and just remove the controllerName field.

Reply:
Okay, added a line in the constructor to explicitly set the variable. I opted to let controller-runtime pick the default instead of choosing our own default.